### PR TITLE
Change params to config

### PR DESCRIPTION
**Pull request by nvnieuwk**
This PR removes all parameters used by the plugin (with the exception of `--help`) and replaces them with configuration values:

| Old parameter | New config option(s) |
|---------------|----------------------|
| `--validationMonochromeLogs <boolean>` | `validation.monochromeLogs = <boolean>` |
| `--validationLenientMode <boolean>` | `validation.lenientMode = <boolean>` |
| `--validationFailUnrecognisedParams <boolean>` | `validation.failUnrecognisedParams = <boolean>` |
| `--validationShowHiddenParams <boolean>` | `validation.showHiddenParams = <boolean>` |
| `--validationIgnoreParams <string>` | `validation.defaultIgnoreParams = <list>` and `validation.ignoreParams = <list>` (`defaultIgnoreParams` is meant to be used by pipeline developers to set the parameters which should always be ignored. `ignoreParams` is meant for the pipeline user to ignore certain parameters.) |

### CODE REVIEW

**Comment by nvnieuwk** [comment]
Closes #4

File: docs/parameters/help_text.md [hunk 1]
```diff
@@ -63,17 +63,7 @@ nextflow run my_pipeline --help outdir
...
-```bash
-nextflow run my_pipeline --help --validationShowHiddenParams
-```
-
-or specifying it in the configuration file
-
-```groovy
-params.validationShowHiddenParams = true
-```
+To show these parameters, set the `validation.showHiddenParams = true` configuration option
```
Comment (awgymer) [inline docs/parameters/help_text.md:66]: Out of interest would `--validation.showHiddenParams` work from the command line?

**Review by awgymer — CHANGES_REQUESTED** [review]
Can we add test(s) that check the new validation params are working properly?
In particular the additive behaviour of `defaultIgnoreParams`

File: docs/parameters/help_text.md (hunk 1 above)
Comment (nvnieuwk) [reply docs/parameters/help_text.md:66]: Nope that doesn't work, and as far as I know you can't change config values on the command line (I've tried...)

Comment (nvnieuwk) [reply docs/parameters/help_text.md:66]: This was one that I was a bit conflicted about making a config value, but I really wanted to reduce the amount of custom parameters and `--help` seemed to be the only one that absolutely needed to be a parameter.

Comment (awgymer) [reply docs/parameters/help_text.md:66]: That is sad. But makes sense as I guess nf would struggle to know if a param was for `params` or a config value

Comment (nvnieuwk) [reply docs/parameters/help_text.md:66]: Indeed! Maybe later we can still make a parameter for this value, but I'd rather see first if that's something the community actually wants. Let's get this beast out first :grin:

**Review by awgymer — APPROVED** [review]

**Comment by nvnieuwk** [comment]
Thanks @awgymer :partying_face:

**Pull request merged**